src/Lower.h

```diff
@@ -18,9 +20,9 @@ class IRMutator;
 /** Given a halide function with a schedule, create a statement that
  * evaluates it. Automatically pulls in all the functions f depends
  * on. Some stages of lowering may be target-specific. */
-EXPORT Stmt lower(const std::vector<Function> &output_funcs,
-                                  const std::string &pipeline_name, const Target &t,
-                  const std::vector<IRMutator *> &custom_passes = std::vector<IRMutator *>());
+ EXPORT Stmt lower(const std::vector<Function> &output_funcs, const std::string &pipeline_name,
```

Comment:
I was undecided on this point. The idea here was to restrict the set of things lowering could do to a module, largely motivated by keeping `lower` as mostly an IR to IR transformer. But I think it would be good to move the buffer finding logic in Pipeline into lowering so `lower` would also need to append `Buffers`. And appending more than one `LoweredFunc` might be useful in some instances. In this model, perhaps `lower` should just become a method on Module. We do lose a little bit in that more context is needed to lower IR and the result is not just IR anymore. But that complexity seems to be a legitimate property of how things work rather than artificial.